### PR TITLE
Questions model migration

### PR DESCRIPTION
**Pull request by weeman1337**
The next model migration

### CODE REVIEW

File: db/migrations/Reference.php [hunk 1]
```diff
@@ -33,5 +35,7 @@ protected function references(Blueprint $table, $targetTable, $fromColumn, $setP
             ->references('id')->on($targetTable)
             ->onUpdate('cascade')
             ->onDelete('cascade');
+
+        return $col;
```
Comment (weeman1337) [inline db/migrations/Reference.php]: I'm returning the column definition here.  
For example the question answerer field should be nullable.

File: db/migrations/Reference.php [hunk 2]
```diff
@@ -17,13 +18,14 @@ protected function referencesUser(Blueprint $table, $setPrimary = true)
 
     /**
      * @param Blueprint $table
-     * @param string    $targetTable
-     * @param string    $fromColumn
-     * @param bool      $setPrimary
+     * @param string $targetTable
+     * @param string $fromColumn
+     * @param bool $setPrimary
+     * @return ColumnDefinition
```
Comment (MyIgel) [inline db/migrations/Reference.php]: Hmm, i think that using spaces is better readable but thats another point for a code style discussion ;)

File: src/Models/Question.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    public function answerer(): BelongsTo
+    {
+        return $this->belongsTo(User::class, 'answerer_id');
+    }
+
+    /**
+     * @return Builder
+     */
+    public static function findUnAnswered(): Builder
```
Comment (MyIgel) [inline src/Models/Question.php]: I would name it `findNotAnswered` but thats only my personal preference

File: src/Models/Question.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Engelsystem\Models;
+
+use Engelsystem\Models\User\User;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+/**
+ * @property integer $id
+ * @property integer $enquirer_id
```
Comment (MyIgel) [inline src/Models/Question.php]: I would name that just user?

File: src/Models/Question.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ * @property string $question
```
Comment (MyIgel) [inline src/Models/Question.php]: Or maybe text? That would be my personal preference and is imho better readable (as its shorter) than `$question->question`

File: src/Models/Question.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ * @property string $answer
+ * @property integer $answerer_id
+ * @property-read \Engelsystem\Models\User\User $answerer
+ * @property-read \Engelsystem\Models\User\User $enquirer
+ * @method static Builder|\Engelsystem\Models\Question newModelQuery()
```
Comment (MyIgel) [inline src/Models/Question.php]: These are (now) in the parent model so no need for these methods ;)

File: src/Models/Question.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+/**
+ * @property integer $id
+ * @property integer $enquirer_id
+ * @property string $question
+ * @property string $answer
+ * @property integer $answerer_id
+ * @property-read \Engelsystem\Models\User\User $answerer
```
Comment (MyIgel) [inline src/Models/Question.php]: `use Engelsystem\Models\User\User;` and so on ^^

File: src/Models/User/User.php [hunk 8]
```diff
@@ -116,4 +120,22 @@ public function newsComments(): HasMany
...
+    public function questions(): HasMany
+    {
+        return $this->hasMany(Question::class, 'enquirer_id')
+            ->where('enquirer_id', $this->id);
+    }
+
+    /**
+     * @return HasMany
+     */
+    public function answers(): HasMany
```
Comment (MyIgel) [inline src/Models/User/User.php]: Hmm, would probably use `answeredQurestions()` here as it interferes less with potential "answers" to other stuff like comments?

**Comment by MyIgel** [comment]
Btw, also `src/Controllers/Metrics/Stats.php` needs to be changed accordingly (searching for `Questions` helps ^^)

File: db/migrations/Reference.php (hunk 2 above)
Comment (weeman1337) [reply db/migrations/Reference.php]: I think that as well ✓

File: src/Models/Question.php (hunk 6 above)
Comment (weeman1337) [reply src/Models/Question.php]: ✓

File: src/Models/Question.php (hunk 5 above)
Comment (weeman1337) [reply src/Models/Question.php]: ✓

File: src/Models/Question.php (hunk 4 above)
Comment (weeman1337) [reply src/Models/Question.php]: ✓

File: src/Models/Question.php (hunk 3 above)
Comment (weeman1337) [reply src/Models/Question.php]: Event shorter `unanswered` ;)

File: src/Models/User/User.php (hunk 8 above)
Comment (weeman1337) [reply src/Models/User/User.php]: I'll make this more specific

**Pull request merged**